lightning-invoice/src/utils.rs

```diff
@@ -430,12 +434,19 @@ fn filter_channels(channels: Vec<ChannelDetails>, min_inbound_capacity_msat: Opt
 	// If all channels are private, return the route hint for the highest inbound capacity channel
 	// per counterparty node. If channels with an higher inbound capacity than the
 	// min_inbound_capacity exists, filter out the channels with a lower capacity than that.
+	// Further, if we are connected to our peer for any channels, only return those.
 	filtered_channels.into_iter()
 		.filter(|(_counterparty_id, channel)| {
-			min_capacity_channel_exists && channel.inbound_capacity_msat >= min_inbound_capacity ||
-			!min_capacity_channel_exists
+			if min_capacity_channel_exists {
+				channel.inbound_capacity_msat >= min_inbound_capacity
+			} else { true }
+		})
+		.filter(|(_counterparty_id, channel)| {
+			if online_channel_exists {
+				channel.is_usable
+			} else { true }
 		})
```

Comment:
Hmmm... actually, this won't work. Consider two channels where one is connected but without enough capacity and the other is disconnected but with enough capacity. The first channel will not pass the first filter even though it would pass the second one. The second channel will pass the first filter but not the second one. So no channels will be included.